pulsar/producer_partition.go

```diff
@@ -369,6 +369,16 @@ func (p *partitionProducer) reconnectToBroker() {
 }
 
 func (p *partitionProducer) runEventsLoop() {
+
+	go func() {
+		for {
+			for range p.connectClosedCh {
```

Comment:
I think this is going to leak a go routine since the `p.connectClosedCh` is never closed?